src/App.tsx

```diff
@@ -15,36 +15,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_FIELD_BY_NAME = 'name',
+  SORT_FIELD_BY_LENGTH = 'length',
+  SORT_BY_NONE = '',
+}
+
+interface Goods {
+  sortField: SortType;
+  reversed: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortField, reversed }: Goods) {
+  const copyGoods = [...goods];
+
+  if (sortField) {
+    copyGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_FIELD_BY_NAME:
+          return good1.localeCompare(good2);
+
+        case SortType.SORT_FIELD_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return copyGoods.reverse();
+  }
+
+  return copyGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.SORT_BY_NONE);
+  const [reversed, SetReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortField === SortType.SORT_FIELD_BY_NAME ? '' : 'is-light'}`}
+          onClick={() => setSortField(SortType.SORT_FIELD_BY_NAME)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortField === SortType.SORT_FIELD_BY_LENGTH ? '' : 'is-light'}`}
+          onClick={() => setSortField(SortType.SORT_FIELD_BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${!reversed ? 'is-light' : ''}`}
+          onClick={() => SetReversed(!reversed)}
```

Comment:
and move to an own fucntion
```suggestion
          onClick={() => setIsReversed(prev => !prev)}
```